From: Takashi Iwai <tiwai@suse.de>
Subject: Fix kABI breakage for max_dev_sectors addition to queue_limits
Patch-mainline: Never
References: boo#961263

block-sd-Fix-device-imposed-transfer-length-limits.patch added a
new field max_dev_sectors to struct queue_limits and this also gives
kABI incompatibility.  Since queue_limits is embedded in other
structs, we can't use a standard technique.

However, queue_limits has some holes at the tail and still 4 bytes
are free there.  So just move the new field at the tail to keep the
struct size.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/linux/blkdev.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -286,7 +286,7 @@ struct queue_limits {
 	unsigned long		seg_boundary_mask;
 
 	unsigned int		max_hw_sectors;
-	unsigned int		max_dev_sectors;
+	/*unsigned int		max_dev_sectors;*/ /* moved below */
 	unsigned int		chunk_sectors;
 	unsigned int		max_sectors;
 	unsigned int		max_segment_size;
@@ -308,6 +308,10 @@ struct queue_limits {
 	unsigned char		cluster;
 	unsigned char		discard_zeroes_data;
 	unsigned char		raid_partial_stripes_expensive;
+#ifndef __GENKSYMS__
+	/* moved here due to kABI compatibility */
+	unsigned int		max_dev_sectors;
+#endif
 };
 
 struct request_queue {
